app/scripts/migrations/133.2.ts

```diff
@@ -0,0 +1,57 @@
+import { hasProperty, isObject } from '@metamask/utils';
+import { cloneDeep } from 'lodash';
+
+type VersionedData = {
+  meta: { version: number };
+  data: Record<string, unknown>;
+};
+
+export const version = 133.2;
+
+/**
+ * This migration removes tokens on mainnet with the
+ * zero address, since this is not a valid erc20 token.
+ *
+ * @param originalVersionedData - Versioned MetaMask extension state, exactly
+ * what we persist to disk.
+ * @returns Updated versioned MetaMask extension state.
+ */
+export async function migrate(
+  originalVersionedData: VersionedData,
+): Promise<VersionedData> {
+  const versionedData = cloneDeep(originalVersionedData);
+  versionedData.meta.version = version;
+  transformState(versionedData.data);
+  return versionedData;
+}
+
+function transformState(state: Record<string, unknown>): void {
+  if (
+    !hasProperty(state, 'TokensController') ||
+    !isObject(state.TokensController) ||
+    !isObject(state.TokensController.allTokens)
+  ) {
+    return;
+  }
+
+  const chainIds = ['0x1'];
```

Comment:
It's possible there are invalid tokens on other chains, or with other addresses.  But I'm not sure that the zero address isn't a valid erc20 on every chain, so I'm keeping it minimal.

Since we have the fallback option in the controller, it's not strictly necessary to remove them all.  But it would make our queries more efficient to remove invalid ones we know of.